src/core/server/http/router/route.ts

```diff
@@ -109,11 +109,13 @@ export interface RouteConfigOptions<Method extends RouteMethod> {
    * - true. A user has to have valid credentials to access a resource
    * - false. A user can access a resource without any credentials.
    * - 'optional'. A user can access a resource if has valid credentials or no credentials at all.
-   * Can be useful when we grant access to a resource but want to identify a user if possible.
+   *               Can be useful when we grant access to a resource but want to identify a user if possible.
+   * - 'try'. A user can access a resource with valid, invalid or without any credentials.
+   *          Users with valid credentials will be authenticated
    *
    * Defaults to `true` if an auth mechanism is registered.
    */
-  authRequired?: boolean | 'optional';
+  authRequired?: boolean | 'optional' | 'try';
```

Comment:
I like the idea of keeping the API surface small. Depending on @azasypkin answer, I'd try to left only the only mode. Probably in a follow-up, we can adopt the `try` behavior for `optional`.